src/amo/components/AddonsByAuthorsCard/index.js

```diff
@@ -55,43 +54,13 @@ export class AddonsByAuthorsCardBase extends React.Component<Props> {
   }
 
   componentWillMount() {
-    const { addons, addonType, authorUsernames, forAddonSlug } = this.props;
+    const { addonType, authorUsernames, dispatch, errorHandler, forAddonSlug } = this.props;
 
-    if (!addons) {
-      this.dispatchFetchAddonsByAuthors({ addonType, authorUsernames, forAddonSlug });
-    }
-  }
-
-  componentWillReceiveProps({
```

Comment:
I haven't run the code yet, but at first glance this doesn't look right. You've removed a bunch of code, and the associated tests, which suggests to me that those use cases are no longer supported. Why is that ok? Have you tested the cases for which you've removed code to make sure they still work?

Please let me know your response to this and then I'll proceed with the rest of the review.

Reply:
Hey @bobsilverberg - sorry I wasn’t too clear above. Basically tho this component (`AddonsByAuthorsCard`) gets remounted so when it runs through componentWillReceiveProps, the `nextProps` are never different from the `this.props`. This means the fetch logic located within this conditional - which compares old and new props - never gets reached. This is why it wasn't refetching the data..

I see the code was extracted out recently from Addons ([code update](https://github.com/mozilla/addons-frontend/commit/21a62088218a91f8f3c5527f451903fccf887202#diff-4953c6065645080e8ab07f343314c395))  and checking old vs newProps worked there but again b/c of remounting it doesn't when moved into its own component. 

It’s a big component but I can look into why Addons is remounting this component. I know I can prevent with shouldComponentMount but not sure if will give us what we need either (or its ramifications here). Anyway maybe we can discuss more tomorrow on a call to sort this out and make sure I move forward with the best approach? :)

Regarding tests...I thought I was removing the tests on this check and moving into the mount test. I always keep `yarn test` running to make sure nothing I didn’t break anything altho maybe it sounds like maybe I need to do more..

Reply:
So I don't think we can do this because if you use the browser "back" button between different addon detail pages, `componentWillReceiveProps()` will definitely be called and we should be using it.

Reply:
Ahh..that's something i didn't see/test. thanks for catching that. I will add that back in here and return those tests